### PR TITLE
Add validateEndpoint method to WavefrontClient.Builder (#169)

### PR DESCRIPTION
**Pull request by attack**
* Add validateEndpoint method to WavefrontClient.Builder
* add javadoc

### CODE REVIEW

File: src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java [hunk 1]
```diff
@@ -237,6 +240,32 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
...
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+      URL url = null;
+
+      try {
+        url = new URL(this.server);
+      } catch (MalformedURLException e) {
+        throw new IllegalStateException(e);
```
Comment (sushantdewan123) [inline src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java:255]: pass in detail msg along with the exception
throw new IllegalStateException("Please provide valid Wavefront endpoint", e);

File: pom.xml [hunk 2]
```diff
@@ -148,6 +148,12 @@
             <version>5.3.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.sun.net.httpserver</groupId>
+            <artifactId>http</artifactId>
+            <version>20070405</version>
```
Comment (basilisk487) [inline pom.xml:154]: @sushantdewan123 Are we clear to depend on this component (licensing-wise)? This looks like it was part of Java SE 6.

File: src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java [hunk 3]
```diff
@@ -237,6 +240,33 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
...
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+      URL url = null;
+
+      try {
+        url = new URL(this.server);
+      } catch (MalformedURLException e) {
+        throw new IllegalStateException("Please provide a reachable Wavefront endpoint", e);
```
Comment (basilisk487) [inline src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java:256]: semantically, IllegalArgumentException fits better here; and error message should be less confusing - something like `this.server + " is not a valid URL"` would work better

File: src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java [hunk 4]
```diff
@@ -237,6 +240,33 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
...
+      }
+
+      try {
+        HttpURLConnection urlConn = (HttpURLConnection) url.openConnection();
+        urlConn.connect();
+        urlConn.disconnect();
+      } catch (IOException e) {
+        throw new IllegalStateException(e);
```
Comment (basilisk487) [inline src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java:264]: ditto

**Review by basilisk487 — CHANGES_REQUESTED** [review]
a couple of minor comments. 

also, for future PRs, please fork the repo and keep the public branches clean

File: src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java (hunk 4 above)
Comment (basilisk487) [reply src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java:264]: And now _this_ is the place to provide details like `"Unable to connect to " + this.server`

File: src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java (hunk 3 above)
Comment (attack) [reply src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java:256]: I agree. This exception was requested, but I can change it.

File: pom.xml (hunk 2 above)
Comment (attack) [reply pom.xml:154]: I am happy to change this out. @basilisk487, any suggestions for a replacement? It just needs to allow connections in tests.

**Review by basilisk487 — APPROVED** [review]

**Review by sushantdewan123 — APPROVED** [review]

**Pull request merged**